uaclient/files/notices.py

```diff
@@ -160,6 +160,56 @@ def remove(self, notice_details: Notice):
         )
         system.ensure_file_absent(os.path.join(directory, filename))
 
+    def _get_notice_file_names(self, directory: str) -> List[str]:
+        """Gets the list of notice file names in the given directory.
+
+        :param directory: The directory to search for notice files.
+        :returns: List of notice file names.
+        """
+        return [
+            file_name
+            for file_name in os.listdir(directory)
+            if os.path.isfile(os.path.join(directory, file_name))
+            and self._is_valid_notice_file(file_name)
+        ]
+
+    def _is_valid_notice_file(self, file_name: str) -> bool:
+        """Checks if the notice file is valid.
+
+        :param file_name: The name of the notice file.
+        :returns: True if the file is valid, False otherwise.
+        """
+        try:
+            order_id, label = file_name.split("-")
+            for n in Notice:
+                if n.order_id == order_id and n.label == label:
+                    return True
+            return False
+        except ValueError:
+            return False
```

Comment:
We can make this more restrictive by using the `is_permanent` field. permanent notices should only appear under `/var/lib/...` and non-permanent notices should only appear under `/run/...`

Reply:
@orndorffgrant Should be fixed and ready for review